### PR TITLE
feat(cb2-7273): Trigger Plate Generation

### PR DESCRIPTION
**Pull request by leifkemp-bjss**
## CB2-7273: Trigger Plate Generation

_PR to allow users to click a link to trigger plate generation for current HGV and TRL records_
[CB2-7273](https://dvsa.atlassian.net/browse/CB2-7273)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by tom-bjss — APPROVED** [review]

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.html [hunk 1]
```diff
@@ -13,6 +13,15 @@
           (submitChange)="handleSubmit()"
         ></app-edit-tech-record-button>
 
+        <ng-container
```
Comment (tom-bjss) [inline src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.html:16]: Need to add role based access here though to Tech.Amend

Comment (tom-bjss) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.html:16]: This will hide the button

Comment (leifkemp-bjss) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.html:16]: I was running into trouble with having both the ngIf and role required being in the container, so I put the role required instead in the button, does that work?

File: src/app/features/tech-record/components/tech-record-generate-plate/tech-record-generate-plate.component.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    this.router.navigate(['..'], { relativeTo: this.route });
+  }
+
+  handleSubmit(plateReasonForGenerating: string): void {
+    this.globalErrorService.clearErrors();
+    if (plateReasonForGenerating === '') {
+      return this.globalErrorService.addError({ error: 'Reason for generating plate is required', anchorLink: 'plateReasonForGenerating' });
+    }
+
+    console.log(plateReasonForGenerating);
```
Comment (naathanbrown) [inline src/app/features/tech-record/components/tech-record-generate-plate/tech-record-generate-plate.component.ts:90]: Reminder to clean this up!

File: src/app/services/technical-record/technical-record.service.ts [hunk 3]
```diff
@@ -294,6 +294,11 @@ export class TechnicalRecordService {
       });
   }
 
+  generatePlate(techRecord?: TechRecordModel) {
+    console.log('Ping.');
```
Comment (naathanbrown) [inline src/app/services/technical-record/technical-record.service.ts:298]: Can you just place a TODO comment here saying we need to come back and add an actual API call

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.html (hunk 1 above)
Comment (naathanbrown) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.html:16]: check tech-record.component,html at the top there for an example where the same logic is applied :)

**Review by naathanbrown — APPROVED** [review]

**Review by naathanbrown — APPROVED** [review]

**Review by tom-bjss — APPROVED** [review]

**Pull request merged**